xhype/xhype/src/linux.rs

```diff
@@ -96,3 +120,218 @@ pub struct SetupHeader {
     handover_offset: u32,
     kernel_info_offset: u32,
 }
+
+const HEADER_OFFSET: u64 = 0x01f1;
+const HDRS: u32 = 0x53726448;
+const ENTRY_64: usize = 0x200;
+
+const LOW64K: usize = 64 * KiB;
+const LOW_MEM_SIZE: usize = 16 * MiB;
+const RD_BASE: usize = 16 * MiB;
+
+pub fn load_linux64(
+    vm: &Arc<RwLock<VirtualMachine>>,
+    kernel_path: String,
+    rd_path: Option<String>,
+    cmd_line: String,
+    mem_size: usize,
+) -> Result<Vec<GuestThread>, Error> {
+    // first we make sure that the kernel is not too old
+    let kn_meta = metadata(&kernel_path)?;
+    let mut kernel_file = File::open(&kernel_path)?;
+    let header: SetupHeader = {
+        let mut buff = [0u8; size_of::<SetupHeader>()];
+        kernel_file.seek(SeekFrom::Start(HEADER_OFFSET))?;
+        kernel_file.read_exact(&mut buff)?;
+        unsafe { transmute(buff) }
+    };
+    if header.setup_sects == 0
```

Comment:
This kind of error checking can drive me a little crazy. I'd prefer something like this

if header.setup_sects = 0 {
err += Err("setup_sects is 0 and that is not allowed") // or however you wrap multiple errors, I forget
}
if head.boot_flag 
etc.etc

and then at the end, return the error, if non-empty, containing ALL these errors. When a failure like this occurs, it is easiest for me if I see ALL the reasons, not that "something was wrong" or just seeing one of many things that are wrong.